easybuild/easyblocks/generic/configuremake.py

```diff
@@ -67,6 +67,101 @@
 DEFAULT_INSTALL_CMD = 'make install'
 
 
+def check_config_guess(config_guess, log=None):
+    """Check timestamp & SHA256 checksum of config.guess script.
+
+    :param configu_gess: Path to config.guess script to check
+    :param log: Instance of a logger to use or None to run in quiet mode
+    :return: Whether the script is valid (matches the version and checksum)
+    """
+    # config.guess includes a "timestamp='...'" indicating the version
+    config_guess_version = None
+    version_regex = re.compile("^timestamp='(.*)'", re.M)
+    res = version_regex.search(read_file(config_guess))
+    if res:
+        config_guess_version = res.group(1)
+
+    config_guess_checksum = compute_checksum(config_guess, checksum_type=CHECKSUM_TYPE_SHA256)
+    try:
+        config_guess_timestamp = datetime.fromtimestamp(os.stat(config_guess).st_mtime).isoformat()
+    except OSError as err:
+        if log:
+            log.warning("Failed to determine timestamp of %s: %s", config_guess, err)
+        config_guess_timestamp = None
+
+    # log version, timestamp & SHA256 checksum of config.guess
+    if log:
+        log.info("config.guess version: %s (last updated: %s, SHA256 checksum: %s)",
```

Comment:
@Flamefire You should introduce a top-level `_log` like we do in other easyblocks, and use that instead?

```python
_log = fancylogger.getLogger('easyblocks.generic.configuremake')

Reply:
The idea was to be able to NOT pass a log to have it be quiet. And I don't understand the global logger: Isn't it supposed to be the logger from the calling class? In this case the rpackage?

Reply:
Why would you want it to be quiet, we're logging useful information here?

Passing a logger is something we should avoid, and since this function isn't part of an easyblock anymore, using a module-level logger is fine. Another option is to create a custom logger in this function, see `pythonpackage.py`

Reply:
You are right. Using a logger where the name is shared between the 2 function which IMO is reasonable as they belong together. One now even calls the other